src/logger.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import time
+import logging
+import datetime
+from enum import Enum
+from abc import ABC
+
+
+class BaseFormatter(ABC, logging.Formatter):
+
+    def formatTime(self, record):
+        """
+        Return the creation time of the specified LogRecord as formatted text.
+        This method should be called from format() by a formatter which
+        wants to make use of a formatted time.
+        It provides an ISO 8601 format with milliseconds.
+        """
+        ct = self.converter(record.created)
+        ct_with_ms = time.mktime(ct) + (record.msecs / 1000)
+        ct_full = datetime.datetime.fromtimestamp(ct_with_ms)
+        return ct_full.isoformat()
+
+    def get_fields(self, record):
+        """
+        Return a dict wth the following fields:
+         - time: ISO 8601 format with milliseconds (from datetime.isoFormat)
+         - level: Log level in uppercase letter
+         - msg: the log message
+         - exception: exception if the logger is specifying exc_info=True
+         - stack: stack information if the logger is specifying stack_info=True
+        """
+        fields = dict()
+        fields['time'] = self.formatTime(record)
+        fields['level'] = record.levelname
+        fields['msg'] = record.getMessage()
```

Comment:
We could chose to align these fields with the ECS standard. That would entail:
time -> @timestamp
level -> log.level
msg -> message

For an example see:
https://www.elastic.co/guide/en/ecs-logging/python/current/installation.html
Full ref:
https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html

Reply:
@schans I understand the need to have a uniform format but I'm not sure ECS is widely used in community. 

So why use more ECS over [Gelf](https://docs.graylog.org/docs/gelf) or [Json Well Format](https://github.com/JSONWellLogFormat/JSONWellLogFormat) ?

From my perspective what's is important is to have a Json format cause all log collectors / aggregators supports JSON. 

I implemented LogFmt as well, cause it's a format widely used in Go community, but I let the maintainers deciding to remove it or not !

Reply:
I guess it's a matter of perspective on what you are used to ;-)

Looking at the Gelf format I wouldn't be a fan of numeric log levels and unix timestamps. 

For the Json Well Format, I'm not sure if that's the one you wanted to link to. It seems specific to (drilling) well measurements in the oil and gas industry.

I don't mind that much about the field names. I think most log aggregation software out there can easily rename and transform. Imho timestamp is a bit more specific than time. If we use msg instead of message you could also opt for ts instead of timestamp. I have seen that plenty as well. But if you prefer to keep the names as they are now that is perfectly fine as well.

Reply:
My reasoning about using `time`, `level` and `msg` is to have fields semantically comprehensible for humans and sufficiently small to limit the size it takes to store the fields in a reversed index (eg. Elasticsearch, Loki...).

That format is what used the golang lib [Logrus](https://github.com/sirupsen/logrus). I saw a lot of Kubernetes projects using `logrus` for logging so considering that project is also for running in Kubernetes I adopt the same log format.

Reply:
Aha. Yes, that makes sense.